### ISSUE TITLE
TestUtil#assertThrows(): support checked exceptions as well

### ISSUE THREAD

**Issue by pyokagan** [se-edu/addressbook-level4#294]
`TestUtil#assertThrows()` currently does not support functions which throw checked exceptions.

e.g.
```java
TestUtil.assertThrows(IllegalValueException.class, () -> { throw new IllegalValueException(""); });
```
does not compile.

It would be useful if it did.

**Comment by pyokagan** [comment]
Marking as low priority since we do not actually need it now. Just putting this down as an issue so that we do not forget it.

**Comment by pyokagan** [comment]
Fixed in 21a1955a4c0b4cd085107a092dec13f66e86af66
